pkg/components/components.go

```diff
@@ -372,63 +372,215 @@ func GetClusterRole() *rbacv1.ClusterRole {
 	return role
 }
 
-func GetCrd() *extv1beta1.CustomResourceDefinition {
-	crd := &extv1beta1.CustomResourceDefinition{
+func GetCrd() *extv1.CustomResourceDefinition {
+	validationSchema := &extv1.CustomResourceValidation{
```

Comment:
Add we sure we cannot store at the ontainer the stuff generated from operator-sdk and read it as golang struct ?

Reply:
we currently can't generate it, it doesn't know how to handle the shared folder:
```
cluster-network-addons-operator (add_csv_schema_validation_detail) $ ./build/_output/bin/operator-sdk generate crds
INFO[0000] Running CRD generator.                       

github.com/kubevirt/cluster-network-addons-operator/pkg/apis/networkaddonsoperator/shared:-: CRD for NetworkAddonsConfig.networkaddonsoperator.network.kubevirt.io has no storage version
FATA[0001] error generating CRDs from APIs in pkg/apis: error generating CRD manifests: error running CRD generator: not all generators ran successfully 
```

plus, (correct me if I'm wrong, but) marshaling the automated yaml to  struct will require us to always know the exact struct format. so in fact it could mean taking up the same amount of time.

Reply:
Well it's all about duplicity I see the description on structs and on the Schema, It has to be doable somehow we will se in the future if we do deeper integration of cnao with operator-sdk.